shell/common/context_options.cc

```diff
@@ -0,0 +1,39 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/shell/common/context_options.h"
+
+#include "flutter/common/graphics/persistent_cache.h"
+
+namespace flutter {
+
+GrContextOptions MakeDefaultContextOptions(ContextType type,
+                                           std::optional<GrBackendApi> api) {
+  GrContextOptions options;
+
+  if (PersistentCache::cache_sksl()) {
+    options.fShaderCacheStrategy = GrContextOptions::ShaderCacheStrategy::kSkSL;
+  }
+  PersistentCache::MarkStrategySet();
+  options.fPersistentCache = PersistentCache::GetCacheForProcess();
+
+  if (api.has_value() && api.value() == GrBackendApi::kOpenGL) {
+    options.fAvoidStencilBuffers = true;
+
+    // To get video playback on the widest range of devices, we limit Skia to
+    // ES2 shading language when the ES3 external image extension is missing.
+    options.fPreferExternalImagesOverES3 = true;
+  }
+
+  // TODO(goderbauer): remove option when skbug.com/7523 is fixed.
+  options.fDisableGpuYUVConversion = true;
+
+  options.fReduceOpsTaskSplitting = GrContextOptions::Enable::kNo;
+
+  options.fReducedShaderVariations = true;
```

Comment:
Guessing this might be the cause of the failure in `EmbedderTest.CanRenderGradientWithCompositor` and others?

Reply:
Yes, I updated the test fixture. I could observe [no noticeable differences](https://github.com/flutter/engine/pull/27016/commits/a6947da2ea9cda57c5386c90edb56f01ac6d35b6?short_path=e6917fe#diff-e6917fed4a9187fe276c282eb54749039d68abfb12539d26ba92509217728bfd) in the results.

Reply:
Also, I did verify that its just because of the shader variations. That is, it was not a side effect of me moving the context option setup into a common translation unit. The test pass if I disable reduced shaders variants.